mozdownload/scraper.py

```diff
@@ -332,10 +335,15 @@ def show_matching_builds(self, builds):
 class DailyScraper(Scraper):
     """Class to download a daily build from the Mozilla server."""
 
-    def __init__(self, branch='mozilla-central', build_id=None, date=None,
+    def __init__(self, branch=None, build_id=None, date=None,
                  build_number=None, revision=None, *args, **kwargs):
         """Create an instance of the daily scraper."""
-        self.branch = branch
+        if not branch:
+            application = kwargs.get('application', None)
```

Comment:
Oh, I see. So this is because we do simply too much in the constructor. I think we should move [line 133](https://github.com/mozilla/mozdownload/blob/master/mozdownload/scraper.py#L133) to `self.download()`, where it makes way more sense.

This code change should fix the problem you are seeing above, and should hopefully not break any tests. Can you please verify that?